### PR TITLE
[ENH] - Update 1d vs 2d (rows & columns) support for array utilities

### PR DESCRIPTION
**Pull request by TomDonoghue**
This relates to #141 - and sweeps through procedures that can work on row / column arrays (both general utilities and space-specific functionality), and works through things to allow them to support either row or column organized 2d arrays. 

The general strategy is to try and infer the orientation of the data, and work from there, meaning that these functions should still work as expected (no extra arguments) on row data, but should also work if the data is column organized. This should maintain backwards compatibility, while supporting columnar data, including supporting position data organized in columns. 

In practice, this means that passing in row position data should still work just as it does currently, but also that passing in column-oriented data should work, and provide the same outputs.

### CODE REVIEW

File: spiketools/tests/spatial/test_occupancy.py [hunk 1]
```diff
@@ -33,6 +33,9 @@ def test_compute_bin_edges():
     position = np.array([[1., 2., 3., 4., 5.], [0., 1., 2., 3., 4.]])
     area_range = [[1, 4], [1, 4]]
 
+    exp_x = np.array([1.0, 2.5, 4.0])
```
Comment (maestapereira) [inline spiketools/tests/spatial/test_occupancy.py:36]: This is very minor, but all other "expected" vars start with `expected`, should we change these ones from `exp_x` and `exp_y` to `expected_x` and  `expected_y`?

File: spiketools/tests/spatial/test_utils.py [hunk 2]
```diff
@@ -7,6 +7,24 @@
 ###################################################################################################
 ###################################################################################################
 
+def test_get_position_xy():
+
+    position_2dr = np.array([[1, 2, 3, 4, 5], [5, 6, 7, 8, 9]])
+    position_2dc = position_2dr.T
+
+    exp_x = np.array([1, 2, 3, 4, 5])
```
Comment (maestapereira) [inline spiketools/tests/spatial/test_utils.py:15]: Same here

File: spiketools/tests/spatial/test_utils.py [hunk 3]
```diff
@@ -20,11 +38,25 @@ def test_compute_nbins():
...
+    # Test 1d position data
+    positions = np.array([1, 2, 3, 4, 5])
     ranges = compute_pos_ranges(positions)
-    ranges[0] == [1, 5]
-    ranges[1] == [5, 9]
+    assert np.array_equal(ranges, np.array([1, 5]))
+
+    # 2d tests
+    positions = np.array([[1, 2, 3, 4, 5], [5, 6, 7, 8, 9]])
+    exp1 = np.array([1, 5])
```
Comment (maestapereira) [inline spiketools/tests/spatial/test_utils.py:48]: And here

**Review by maestapereira — APPROVED** [review]
Looks good! Just had a note on variable naming for the tests!

File: spiketools/tests/spatial/test_occupancy.py (hunk 1 above)
Comment (TomDonoghue) [reply spiketools/tests/spatial/test_occupancy.py]: Yes, good eye for consistency! I have updated all the examples you noted - thank you!

File: spiketools/utils/trials.py [hunk 4]
```diff
@@ -83,3 +85,32 @@ def split_trials_by_condition_array(trials, conditions):
...
+def recombine_trial_data(times_trials, values_trials, axis=None):
+    """Recombine data across trials.
+
+    Parameters
+    ----------
+    times_trials : list of 1d array
+        Epoched timestamps.
+    values_trials : list of 1d or 2d array
+        Epoched trial data.
+    axis : {-1, 0, 1}, optional
```
Comment (claire98han) [inline spiketools/utils/trials.py:99]: What does `-1` mean here?

**Review by claire98han — APPROVED** [review]
@TomDonoghue - All look great to me!

File: spiketools/utils/trials.py (hunk 4 above)
Comment (TomDonoghue) [reply spiketools/utils/trials.py:99]: Good question! I had to go double check lol. 
-1 is used if the the input arrays are specified to be 1d vectors, which don't have an orientation (as defined in numpy). This should be made explicit here, and across the any other mentions of the axis argument.

**Comment by TomDonoghue** [comment]
@claire98han - thanks for checking! Were there other docstrings you noticed that could use updating? If so, please let me know which ones - we might as well fix them now!

**Comment by claire98han** [comment]
> @claire98han - thanks for checking! Were there other docstrings you noticed that could use updating? If so, please let me know which ones - we might as well fix them now!

Sorry I commented on the wrong PR, I meant for #153.

**Comment by TomDonoghue** [comment]
Sounds good! Thanks both for checking this one - I'll merge it in now!

File: spiketools/utils/trials.py (hunk 4 above)
Comment (TomDonoghue) [reply spiketools/utils/trials.py:99]: Wait no, this isn't even correct lol. vector is also 0. -1 is a default answer that indicates that the axis could not be inferred (for example, if given an empty axis). I'm checking through to fix the docs on this now.

**Pull request merged**